src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -4370,8 +4449,22 @@ private static void EmitTimeoutCheckIfNeeded(IndentedTextWriter writer, RegexMet
                 bool negated = RegexCharClass.IsNegated(node.Str) ^ negate;
 
                 Span<char> setChars = stackalloc char[5]; // current max that's vectorized
-                int setCharsCount;
-                if ((setCharsCount = RegexCharClass.GetSetChars(node.Str, setChars)) > 0)
+                int setCharsCount = RegexCharClass.GetSetChars(node.Str, setChars);
+
+                // Prefer IndexOfAnyInRange over IndexOfAny for sets of 2-5 values that fit in a single range
```

Comment:
I'm surprised IndexOfAnyInRange is better than IndexOfAny for two chars.

Reply:
They're pretty close, but at least in https://github.com/dotnet/runtime/pull/78861 I was seeing the range come across faster. Looks like in `main` IndexOfAny(2) is a bit better for short inputs (I removed some overhead for the scalar path for range in that PR).

Reply:
Ok, thanks. If they're relatively close on longer searches, let's prefer the one that's cheaper for shorter searches.